projects/step-frontend/src/lib/modules/execution/components/aggregated-tree-node-history/execution-piechart/aggregated-tree-node-statuses-piechart.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, effect, ElementRef, input, OnDestroy, viewChild } from '@angular/core';
+import Chart from 'chart.js/auto';
+
+export interface TreeNodePieChartSlice {
+  color: string;
+  label: string;
+  count: number;
+}
+
+@Component({
+  selector: 'step-aggregated-tree-node-statuses-piechart',
+  templateUrl: './aggregated-tree-node-statuses-piechart.component.html',
+  styleUrls: ['./aggregated-tree-node-statuses-piechart.component.scss'],
+  standalone: false,
+})
+export class AggregatedTreeNodeStatusesPiechartComponent implements OnDestroy {
+  readonly slices = input.required<TreeNodePieChartSlice[]>();
+  readonly size = input<number>(24);
+  readonly startAngleDeg = input<number>(-90);
+  readonly circumferenceDeg = input<number>(360);
+  readonly emptyColor = input<string>('#e5e7eb');
+  readonly highlight = input<boolean>(false);
+
+  private readonly canvas = viewChild<ElementRef<HTMLCanvasElement>>('canvas');
+  private chart?: Chart | undefined;
+
+  private renderEffect = effect(() => {
+    const el = this.canvas();
+    const sz = this.size();
+    const start = this.startAngleDeg();
+    const circ = this.circumferenceDeg();
+    const empty = this.emptyColor();
+
+    if (!el) return;
+
+    const src = (this.slices() ?? [])
+      .filter((s) => (s?.count ?? 0) > 0)
+      .sort((s1, s2) => s1.label.localeCompare(s2.label));
+    const total = src.reduce((a, s) => a + (s.count || 0), 0);
+
+    const data =
+      total === 0
+        ? {
+            labels: ['empty'],
+            datasets: [{ data: [100], backgroundColor: [empty] }],
+          }
+        : {
+            labels: src.map((s) => s.label),
+            datasets: [
+              {
+                // feed raw counts to Chart.js (it will scale arcs proportionally)
+                data: src.map((s) => s.count),
+                backgroundColor: src.map((s) => s.color),
+              },
+            ],
+          };
+
+    const arcStyle =
```

Comment:
might be fine but I'm unsure what an "Arc" is in this context. Maybe there is a better name?

Reply:
Arc is directly coming from the chartjs structure/terminology. In my language it is also called "circle arc" which is kind of the border of a circle